meteor/client/lib/editAttribute/edit-attribute-multi-select.tsx

```diff
@@ -0,0 +1,65 @@
+import { MultiSelect, MultiSelectOption } from '../multiSelect'
+import ClassNames from 'classnames'
+import * as React from 'react'
+import { EditAttributeBase, IEditAttributeBaseProps, wrapEditAttribute } from './edit-attribute-base'
+
+interface EditAttributeMultiSelectProps extends IEditAttributeBaseProps {
+	options: MultiSelectOption[]
+}
+
+export function EditAttributeMultiSelect(props: EditAttributeMultiSelectProps) {
+	return <WrappedEditAttributeMultiSelect {...props} />
+}
+
+const WrappedEditAttributeMultiSelect = wrapEditAttribute(
+	class EditAttributeMultiSelect extends EditAttributeBase {
+		constructor(props) {
+			super(props)
+
+			this.handleChange = this.handleChange.bind(this)
+		}
+
+		private handleChange(changedOptions: MultiSelectOption[]): void {
+			this.handleUpdate(changedOptions)
+		}
+
+		private getCurrentlySelectedOptions(): MultiSelectOption[] {
+			const attribute = this.getAttribute()
+			if (!attribute || !Array.isArray(attribute)) {
+				return []
+			}
+			return attribute
+		}
+
+		private getAvailableOptions(): MultiSelectOption[] {
+			return (this.props as EditAttributeMultiSelectProps).options
+		}
+
+		private getMissingOptions(availableOptions: MultiSelectOption[]): MultiSelectOption[] {
+			return this.getCurrentlySelectedOptions()
+				.filter((selectedOption) => availableOptions.every((option) => option.value !== selectedOption.value))
+				.map((option) => {
+					return {
+						value: option.value,
+						label: option.label,
+						className: 'option-missing',
+					}
+				})
```

Comment:
You can use shorthand notation for object creation and use argument desctructuring:
```typescript
				.map(({ value, label }) => ({
					value,
					label,
					className: 'option-missing',
				}))
```

Reply:
Updated